src/tests/GfxReplayTest.cpp

```diff
@@ -449,6 +456,79 @@ void GfxReplayTest::testSimulatorIntegration() {
   }
 }
 
+// test lights data by recording and playback through the simulator interface
+void GfxReplayTest::testLightIntegration() {
+  const auto testFilepath =
+      Corrade::Utility::Path::join(DATA_DIR, "./gfx_replay_test.json");
+  const LightInfo pointLight0{Magnum::Vector4{1.5f, 2.0f, 2.5f, 1.0f},
+                              {1.5, 2.0, 5.0},
+                              LightPositionModel::Global};
+  const LightInfo pointLight1{Magnum::Vector4{-10.0f, 4.25f, 10.0f, 1.0f},
+                              {5.0, 5.0, 0.0},
+                              LightPositionModel::Global};
+  const LightSetup lightSetup{pointLight0, pointLight1};
+  const std::string lightSetupKey = "custom_lighting";
+
+  // record a playback file
+  {
+    SimulatorConfiguration simConfig{};
+    simConfig.enableGfxReplaySave = true;
+    simConfig.createRenderer = false;
+    simConfig.enablePhysics = false;
+    auto sim = Simulator::create_unique(simConfig);
+    CORRADE_VERIFY(sim);
+
+    sim->setLightSetup(lightSetup, lightSetupKey);
```

Comment:
How about multiple calls to `setLightSetup` (with differing setups) and `saveKeyframe`? And then we can verify that lights are getting dynamically added and removed properly during playback.

Reply:
I updated the test accordingly.